add go-hit

Eun opened the pull request:
Please check if what you want to add to `awesome-go` list meets [quality standards](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Please provide package links to:**

- github.com repo: https://github.com/Eun/go-hit
- pkg.go.dev: https://pkg.go.dev/github.com/Eun/go-hit
- goreportcard.com: https://goreportcard.com/report/github.com/Eun/go-hit
- coverage service link: https://codecov.io/gh/Eun/go-hit


Very good coverage

**Note**: that new categories can be added only when there are 3 packages or more.

**Make sure that you've checked the boxes below before you submit PR:**
- [x] I have added my package in alphabetical order.
- [x] I have an appropriate description with correct grammar.
- [x] I know that this package was not listed before.
- [x] I have added pkg.go.dev link to the repo and to my pull request.
- [x] I have added coverage service link to the repo and to my pull request.
- [x] I have added goreportcard link to the repo and to my pull request.
- [x] I have read [Contribution guidelines](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#quality-standard).

Thanks for your PR, you're awesome! :+1:

avelino commented:
Thank you for contributing with [awesome-go](https://github.com/avelino/awesome-go), we will revise your contribution as soon as possible.

Automation body links content check:
- godoc.org or pkg.go.dev: True
- goreportcard.com: True
- coverage: True

> your project is about to be approved, it's under revision, _it may take a few days_

Eun commented:
Thanks for feedback I switched to coveralls.io because codecov reported some invalid data.

I also improved test coverage and fixed the linter errors.

Here are the new urls:
coverage service link: https://coveralls.io/github/Eun/go-hit?branch=master
pkg.go.dev: https://pkg.go.dev/github.com/Eun/go-hit@v0.5.4
